operate/webapp/src/main/java/io/camunda/operate/webapp/security/permission/PermissionsService.java

```diff
@@ -198,7 +198,9 @@ private List<Long> getAuthenticatedUserRoleKeys() {
     if (requestAuthentication != null) {
       final Object principal = requestAuthentication.getPrincipal();
       if (principal instanceof final CamundaUser authenticatedPrincipal) {
```

Comment:
❓ What about the `CamundaOidcUser` and clients?

Reply:
These instanceof checks all had to be changed to check for CamundaPrincipal instead. Thanks for catching that.